catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/QueryApplication.java

```diff
@@ -189,4 +209,34 @@ public void setFeatureService(FeatureService featureService) {
   public void setEndpointUtil(EndpointUtil util) {
     this.util = util;
   }
+
```

Comment:
Not sure that we need to union properties and take first non-zero hit count.
The reduction overwrites previous properties with new values if any. Otherwise, the reduction aggregates properties. Unless there is a good reason to aggregate properties, I think we'd be fine with taking the first, last or middle properties and using them.
Why use the first non-zero value of hits? The hit count should be the same for all query result objects. Only the actual results should change.

Reply:
Is there an answer for Aaron's question?